builtin/providers/google/resource_google_project.go

```diff
@@ -55,36 +88,57 @@ func resourceGoogleProject() *schema.Resource {
 	}
 }
 
-// This resource supports creation, but not in the traditional sense.
-// A new Google Cloud Project can not be created. Instead, an existing Project
-// is initialized and made available as a Terraform resource.
 func resourceGoogleProjectCreate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
 
-	project, err := getProject(d, config)
+	var pid string
+	var err error
+	pid = d.Get("project_id").(string)
+	if pid == "" {
+		pid, err = getProject(d, config)
+		if err != nil {
+			return fmt.Errorf("Error getting project ID: %v", err)
+		}
+	}
+	if pid == "" {
+		return fmt.Errorf("One of 'project_id' or 'id' must be set in the config")
```

Comment:
I'd remove `or 'id'` from this--we should try to discourage any new use of `id`.